main.go

```diff
@@ -30,7 +31,14 @@ func main() {
 
 	// TODO(runcom): parametrize this when the bin starts
 	h := authz.NewHandler(novolume)
-	if err := h.ServeUnix("root", *flPluginSocket); err != nil {
-		logrus.Fatal(err)
+
+	fs := activation.Files(false)
+	if len(fs) > 1 {
+		logrus.Fatalf("expected one fd from systemd, got %d", len(fs))
+	} else if len(fs) == 1 {
+		err = h.ServeUnix("root", *flPluginSocket, fs[0])
+	} else {
+		err = h.ServeUnix("root", *flPluginSocket, nil)
 	}
```

Comment:
@rhatdan the third parameter is for the systemd socket which is passed  otherwise it's not replying on the first call - and yes it works if not socket activated